.github/workflows/release_crypto_js.yml

```
@@ -0,0 +1,56 @@
+# This workflow releases the `matrix-sdk-crypto-js` project.
+#
+# It is triggered when a new tag matches
+# `matrix-sdk-crypto-js-v[0-9]+.*`, which then build the package for
+# the binding, run its tests to ensure everything is still correct,
+# and publish the package on NPM.
+
+
+name: Release `crypto-js`
+
+env:
+  CARGO_TERM_COLOR: always
+  PKG_PATH: "bindings/matrix-sdk-crypto-js"
+
+on:
+  push:
+    tags:
+      - matrix-sdk-crypto-js-v[0-9]+.*
+
+jobs:
+  publish-matrix-sdk-crypto-js:
+    name: Publish 🕸 [m]-crypto-js
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+
+      - name: Load cache
+        uses: Swatinem/rust-cache@v1
+
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.0
+
+      - name: Install NPM dependencies
+        working-directory: ${{ env.PKG_PATH }}
+        run: npm install
+
+      - name: Configure NPM auth token
+        working-directory: ${{ env.PKG_PATH }}
+        run: npm set "//registry.npmjs.org/:_authToken" "${{ secrets.NPM_TOKEN }}"
+
```

what happened to adding the package to the github release? Also, you've added a `prepublish` cmd but that isn't used. why not?

```suggestion

      - name: Package the bindings
        working-directory: ${{ env.PKG_PATH }}
        run: npm run prepublish

      - name: Upload npm package to github release
        uses: softprops/action-gh-release@v1
        with:
          files: ${{env.PKG_PATH}}/*tgz
```
(not sure if the `files` might need to be in a subdirectory...

---

A “pre<script>” is run when running “<script>”. That's weird I know…

Quoting [the official documentation for `"scripts"`](https://docs.npmjs.com/cli/v8/using-npm/scripts):

> _Pre_ and _post_ commands with matching names will be run for those as well (e.g. `premyscript`, `myscript`, `postmyscript`)

---

> what happened to adding the package to the github release?

I've forgot about that. Do we want to support that? Do we know if people will install an NPM package from a Github release?

---

geee ... npm, and all its magic...

I don't think people install from such a release, but if we already put up a tag and release on github, I think it is fair to expect that there is a build tarball either way ... even if only to confirm the npm version and this one match up and haven't been tempered with...

---

OK, I'm adding it :-).

---

Done.